dartagnan/src/main/java/com/dat3m/dartagnan/expression/BExpr.java

```diff
@@ -1,22 +1,26 @@
 package com.dat3m.dartagnan.expression;
 
-import com.microsoft.z3.Context;
-import com.microsoft.z3.Expr;
-import com.microsoft.z3.Model;
-
 import java.math.BigInteger;
 
+import org.sosy_lab.java_smt.api.Formula;
+import org.sosy_lab.java_smt.api.FormulaManager;
+import org.sosy_lab.java_smt.api.Model;
+import org.sosy_lab.java_smt.api.SolverContext;
+
 import com.dat3m.dartagnan.program.event.Event;
 
 public abstract class BExpr implements ExprInterface {
 
     @Override
-    public Expr toZ3Int(Event e, Context ctx) {
-        return ctx.mkITE(toZ3Bool(e, ctx), ctx.mkInt(1), ctx.mkInt(0));
+    public Formula toIntFormula(Event e, SolverContext ctx) {
+    	FormulaManager fmgr = ctx.getFormulaManager();
+		return fmgr.getBooleanFormulaManager().ifThenElse(toBoolFormula(e, ctx), 
+				fmgr.getIntegerFormulaManager().makeNumber(BigInteger.ONE), 
+				fmgr.getIntegerFormulaManager().makeNumber(BigInteger.ZERO));
```

Comment:
refactor fmgr.getIntegerFormulaManager into a local variable

Reply:
About to be pushed